### DISCUSSION TITLE
Jobs fired sequential or asynchornous?

### DISCUSSION THREAD

**Discussion by abishz17** [go-co-op/gocron#758] [General]
Are jobs fired concurrently (asynchronously) by default, where each job runs independently in its own goroutine?

**Comment by JohnRoesler** [comment]
hi @abishz17 each job is scheduled sequentially initially by the scheduler. Being scheduled, means the job is spun off into a goroutine where it waits on a timer until it should run. At that point, the job makes a call to the executor to have the actual job function run. The executor runs the job in a separate goroutine.

TL;DR - Yes, they are run async.
